Formula/bazel.rb

```diff
@@ -29,24 +30,37 @@ def install
     # Force Bazel ./compile.sh to put its temporary files in the buildpath
     ENV["BAZEL_WRKDIR"] = buildpath/"work"
     # Force Bazel to use openjdk@11
-    ENV["JAVA_HOME"] = Formula["openjdk@11"].opt_libexec/"openjdk.jdk/Contents/Home"
     ENV["EXTRA_BAZEL_ARGS"] = "--host_javabase=@local_jdk//:jdk"
+    on_macos { ENV["JAVA_HOME"] = Formula["openjdk@11"].opt_libexec/"openjdk.jdk/Contents/Home" }
+    on_linux { ENV["JAVA_HOME"] = Formula["openjdk@11"].opt_libexec }
+    # Force Bazel to use Homebrew python
+    ENV.prepend_path "PATH", Formula["python@3.9"].opt_libexec/"bin"
+
+    # Bazel clears environment variables other than PATH during build, which
+    # breaks Homebrew shim scripts. We don't see this issue on macOS since
+    # the build uses a Bazel-specific wrapper for clang rather than the shim;
+    # specifically, it uses `external/local_config_cc/wrapped_clang`.
+    #
+    # The workaround here is to disable the Linux shim for C/C++ compilers.
+    # Remove this when a way to retain HOMEBREW_* variables is found.
+    on_linux do
+      ENV["CC"] = "/usr/bin/cc"
+      ENV["CXX"] = "/usr/bin/c++"
+    end
```

Comment:
If we're going to avoid the shims we should probably just do `env :std` and add an allowlist for formulae that can do this.

Reply:
That sounds like overkill to me.

Reply:
This is probably last decision to make before merge.

`bazel`'s feature is to provide reproducible builds and part of that is to clear out the environment. Since `bazel` is built with `bazel`, we need to work around this.

I haven't taken a look at full impact of using `env :std`, so I would like some additional opinions on that.

Currently, `bazel` formula build on Linux only fails due to compiler shims, which allows us to avoid issue by just modifying `CC`/`CXX`.

Reply:
I'm in favor of just using the workaround you have here for now.  I think we may have to consider treating formula which are built with `bazel` or `bazelisk` the same way we treat formulae that depend on `scons` right now (which does use `env :std`), but for for building `bazel` itself I don't think `env :std` would be needed.